external/webserver/http_client.c

```diff
@@ -103,6 +105,29 @@ pthread_addr_t http_handle_client(pthread_addr_t arg)
 		} else {
 			HTTP_LOGD("Client %d  in normal case.\n", sock_fd);
 		}
+#else
+		do {
+			http_keyvalue_list_init(&request_params);
+			result = http_recv_and_handle_request(p, &request_params);
+			http_keyvalue_list_release(&request_params);
+
+			HTTP_LOGD("Client %d in keep-alive %d.\n", p->keep_alive);
+
+			if (result != HTTP_OK) {
+				HTTP_LOGD("Client %d  in error case.\n", sock_fd);
+				break;
+			} else {
+				HTTP_LOGD("Client %d  in normal case.\n", sock_fd, p->keep_alive);
+
+				// Close client socket if connection is not persistent one.
+				if (p->keep_alive == 0) {
+					HTTP_LOGD("Client %d closing.\n", sock_fd);
+					close(p->client_fd);
```

Comment:
Here client_fd is closed, but it doesn't have to close it after main loop finished? I mean line 130.